diff --git a/lib/index.js b/lib/index.js
index 76feb0ffe34ebaa48d410ce9c9ea0c93154ce907..1bdf19b3a1ca1dd36a64393523625c82b6b56af2 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -41,7 +41,7 @@ const d = null, {min: a, max: h, abs: p, floor: g} = Math, v = (e, t, o) => a(o,
     return v(r, 0, e.l - 1);
 }, I = (e, t, o) => {
     const n = t - e.l;
-    return e.i = o ? -1 : a(t - 1, e.i), e.l = t, n > 0 ? ($(e.u, n), $(e.t, n, o), 
+    return e.i = o ? -1 : a(t - 1, e.i), e.l = t, n > 0 ? ($(e.u, n), $(e.t, n, o),
     e.o * n) : (e.u.splice(n), (o ? e.t.splice(0, -n) : e.t.splice(n)).reduce((t, o) => t - (-1 === o ? e.o : o), 0));
 }, k = "undefined" != typeof window, R = e => e.documentElement, T = e => e.ownerDocument, C = e => e.defaultView, M = /*#__PURE__*/ S(() => !!/iP(hone|od|ad)/.test(navigator.userAgent) || "MacIntel" === navigator.platform && navigator.maxTouchPoints > 0), O = /*#__PURE__*/ S(() => "scrollBehavior" in R(document).style), E = e => h(e.h(), e.p()), H = (e, t = 40, o = 0, n, r = !1) => {
     let i = !!o, s = 1, l = 0, c = 0, f = 0, u = 0, g = 0, v = 0, _ = 0, w = 0, S = d, k = [ 0, i ? h(o - 1, 0) : -1 ], R = 0, T = !1;
@@ -77,7 +77,7 @@ const d = null, {min: a, max: h, abs: p, floor: g} = Math, v = (e, t, o) => a(o,
             let t, o;
             if (v) [t, o] = k; else {
                 let n = h(0, H()), i = n + l;
-                r || (e = h(0, e), 1 !== _ && (n -= e), 2 !== _ && (i += e)), [t, o] = k = W(h(0, n), h(0, i)), 
+                r || (e = h(0, e), 1 !== _ && (n -= e), 2 !== _ && (i += e)), [t, o] = k = W(h(0, n), h(0, i)),
                 S && (t = a(t, S[0]), o = h(o, S[1]));
             }
             return [ h(t, 0), a(o, C.l - 1) ];
@@ -126,7 +126,7 @@ const d = null, {min: a, max: h, abs: p, floor: g} = Math, v = (e, t, o) => a(o,
                         let n;
                         if (2 === w) n = !0; else if (S && 1 === w) n = t < S[0]; else {
                             const e = E(), o = J(t), r = A(t);
-                            n = 1 !== _ && 0 === w ? o + r < e : o < e && o + r < e + l;
+                            n = 1 !== _ && 0 === w ? o + r <= e : o < e && o + r < e + l; // See https://github.com/inokawa/virtua/pull/894
                         }
                         return n && (e += o - A(t)), e;
                     }, 0));
@@ -205,7 +205,7 @@ const d = null, {min: a, max: h, abs: p, floor: g} = Math, v = (e, t, o) => a(o,
         passive: !0
     }), {
         A: () => {
-            t.removeEventListener("scroll", p), t.removeEventListener("wheel", g), t.removeEventListener("touchstart", v), 
+            t.removeEventListener("scroll", p), t.removeEventListener("wheel", g), t.removeEventListener("touchstart", v),
             t.removeEventListener("touchend", m), h.J();
         },
         L: () => {
@@ -430,13 +430,13 @@ const d = null, {min: a, max: h, abs: p, floor: g} = Math, v = (e, t, o) => a(o,
     let o;
     const n = new WeakMap, r = new Set, i = new Set, s = new Map, l = (e, t) => `${e}-${t}`, c = V(c => {
         const f = new Set, u = new Set;
-        for (const {target: r, contentRect: {width: i, height: d}} of c) if (r.offsetParent) if (r === o) e.B(4, d), 
+        for (const {target: r, contentRect: {width: i, height: d}} of c) if (r.offsetParent) if (r === o) e.B(4, d),
         t.B(4, i); else {
             const e = n.get(r);
             if (e) {
                 const [t, o] = e, n = l(t, o), r = s.get(n);
                 let c, a;
-                r ? (r[0] !== d && (c = !0), r[1] !== i && (a = !0)) : c = a = !0, c && f.add(t), 
+                r ? (r[0] !== d && (c = !0), r[1] !== i && (a = !0)) : c = a = !0, c && f.add(t),
                 a && u.add(o), (c || a) && s.set(n, [ d, i ]);
             }
         }
